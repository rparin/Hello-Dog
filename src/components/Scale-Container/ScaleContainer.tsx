import { cn } from "@/lib/utils";
import Progressbar from "./Progressbar";
import { barType } from "./Progressbar";

export default function ScaleContainer({
  className,
  title,
  scales,
}: {
  className?: string;
  title: string;
  scales: Array<barType>;
}) {
  const barItems = scales.map((item) => {
    return (
      <Progressbar
        className="overflow-auto md:mb-1"
        key={item.name}
        barValue={item}
      ></Progressbar>
    );
  });

  return (
    <div
      className={cn(
        "flex h-full w-full flex-col items-center bg-accent text-black",
        className,
      )}
    >
      <h1 className="text-sm leading-none md:text-xl">{title}</h1>
      {barItems}
    </div>
  );
}
