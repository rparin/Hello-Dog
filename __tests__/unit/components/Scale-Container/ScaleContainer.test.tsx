import { render } from "@testing-library/react";
import { axe, toHaveNoViolations } from "jest-axe";
import ScaleContainer from "@/components/Scale-Container/ScaleContainer";

expect.extend(toHaveNoViolations);

test("should have no accessibility violations", async () => {
  const { container } = render(
    <ScaleContainer
      title={"Good With"}
      scales={[{ name: "Energy", value: 3 }]}></ScaleContainer>
  );
  const results = await axe(container);
  expect(results).toHaveNoViolations();
});
