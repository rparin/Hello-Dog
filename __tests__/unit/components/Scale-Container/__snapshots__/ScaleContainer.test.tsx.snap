// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders component unchanged 1`] = `
<div>
  <div
    class="flex h-full w-full flex-col items-center bg-accent text-black"
  >
    <h1
      class="text-base leading-none md:text-xl"
    >
      Good With
    </h1>
    <div
      class="hide-scrollbar flex h-full w-full flex-col overflow-auto"
    >
      <div
        class="flex h-full w-full flex-col md:pb-1"
      >
        <p
          class="ml-1.5 mr-1.5 border-b border-solid border-black text-xs leading-none md:text-base"
        >
          Energy
        </p>
        <div
          class="mx-1 mt-0.5 h-full max-h-3 min-h-2 rounded-3xl bg-white md:mx-0.5 md:max-h-4 lg:max-h-5"
        >
          <div
            class="h-full rounded-3xl w-3/5 bg-[#E59E24]"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
